From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: thomasclark <tjclark@wpi.edu>
Date: Thu, 24 Jul 2014 13:23:09 -0400
Subject: [PATCH 0194/6262] CANJaguar::ChangeControlMode marks the controlmode
 as unverified

This previous caused a bug where the Jaguar would sometimes
behave as if it were in the wrong mode, even when everything was
verfied.

Change-Id: Id5b96e0f7e64066eaaa0e5be4ba53fca76ba0703
---
 wpilibc/wpilibC++/lib/CANJaguar.cpp                              | 1 +
 .../src/main/java/edu/wpi/first/wpilibj/CANJaguar.java           | 1 +
 2 files changed, 2 insertions(+)

diff --git a/wpilibc/wpilibC++/lib/CANJaguar.cpp b/wpilibc/wpilibC++/lib/CANJaguar.cpp
index 737e942129f234c30c660b7e5831955f9bd21eeb..fee23da5537188d5a7ac77cd482d6b3d6a64d61b 100644
--- a/wpilibc/wpilibC++/lib/CANJaguar.cpp
+++ b/wpilibc/wpilibC++/lib/CANJaguar.cpp
@@ -1617,6 +1617,7 @@ void CANJaguar::ChangeControlMode(ControlMode controlMode)
 
 	// Update the local mode
 	m_controlMode = controlMode;
+	m_controlModeVerified = false;
 
 	HALReport(HALUsageReporting::kResourceType_CANJaguar, m_deviceNumber, m_controlMode);
 }
diff --git a/wpilibj/wpilibJavaDevices/src/main/java/edu/wpi/first/wpilibj/CANJaguar.java b/wpilibj/wpilibJavaDevices/src/main/java/edu/wpi/first/wpilibj/CANJaguar.java
index 722b36282211c098db29e6890fd8f9c603a696ff..d8c92e253532591bdf4264a688c1210495776bcb 100644
--- a/wpilibj/wpilibJavaDevices/src/main/java/edu/wpi/first/wpilibj/CANJaguar.java
+++ b/wpilibj/wpilibJavaDevices/src/main/java/edu/wpi/first/wpilibj/CANJaguar.java
@@ -1350,6 +1350,7 @@ public class CANJaguar implements MotorSafety, PIDOutput, SpeedController, LiveW
 
 		// Update the local mode
 		m_controlMode = controlMode;
+		m_controlModeVerified = false;
 	}
 
 	/**
